test/Microsoft.AspNet.Razor.Runtime.Test/TagHelpers/TagHelperDescriptorFactoryTest.cs

```diff
@@ -1564,6 +1796,59 @@ public void ValidateTagHelperAttributeDescriptor_WithValidPrefix_ReturnsTrue(str
             return data;
         }
 
+        [EditorBrowsable(EditorBrowsableState.Always)]
+        private class DefaultEditorBrowsableTagHelper : TagHelper
+        {
+            [EditorBrowsable(EditorBrowsableState.Always)]
+            public int Property { get; set; }
+        }
+
+        private class HiddenPropertyEditorBrowsableTagHelper : TagHelper
+        {
+            [EditorBrowsable(EditorBrowsableState.Never)]
+            public int Property { get; set; }
+        }
+
+        private class MultiHiddenPropertyEditorBrowsableTagHelper : TagHelper
+        {
+            [EditorBrowsable(EditorBrowsableState.Never)]
+            public int Property { get; set; }
+
+            public virtual int Property2 { get; set; }
+        }
+
+        private class OverriddenPropertyEditorBrowsableTagHelper : MultiHiddenPropertyEditorBrowsableTagHelper
+        {
+            [EditorBrowsable(EditorBrowsableState.Never)]
+            public override int Property2 { get; set; }
+        }
+
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        private class EditorBrowsableTagHelper : TagHelper
+        {
+            [EditorBrowsable(EditorBrowsableState.Never)]
+            public virtual int Property { get; set; }
+        }
+
+        private class InheritedEditorBrowsableTagHelper : EditorBrowsableTagHelper
+        {
+            public override int Property { get; set; }
+        }
+
+        [EditorBrowsable(EditorBrowsableState.Advanced)]
+        private class OverriddenEditorBrowsableTagHelper : EditorBrowsableTagHelper
+        {
+            [EditorBrowsable(EditorBrowsableState.Advanced)]
+            public override int Property { get; set; }
+        }
+
+        [TargetElement("p")]
+        [TargetElement("div")]
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        private class MultiEditorBrowsableTagHelper : TagHelper
```

Comment:
Hu? It's just the name of the test class.

- Multi = more than 1 element
- EditorBrowsable = Has the `EditorBrowsableAttribute`